plugin/torch/torch_base.cc

```diff
@@ -7,38 +7,42 @@
 #include "./torch_base.h"
 
 namespace mxnet {
-lua_State* TorchState::LuaState() {
-  thread_local lua_State* state = NULL;
-  if (!state) {
-    state = luaL_newstate();
-    luaL_openlibs(state);
-    luaL_loadstring(state,
-                    "require 'torch'\n"
-                    "require 'nn'\n"
+lua_State* TorchState::NewLuaState() {
```

Comment:
It's better to make functions in TorchState non-static and store the lua_state in TorchState.
Can be done latter though.